include/deal.II/base/index_set.h

```diff
@@ -1723,7 +1723,7 @@ IndexSet::add_indices(const ForwardIterator &begin, const ForwardIterator &end)
       size_type       end_index   = begin_index + 1;
       ForwardIterator q           = p;
       ++q;
-      while ((q != end) && (*q == end_index))
+      while ((q != end) && (size_type(*q) == end_index))
```

Comment:
why not do the following?
```suggestion
      while ((q != end) && (static_cast<size_type>(*q) == end_index))
```

Reply:
Yes, I think the explicit cast is what you want.